plugins/amongus/amongus.cpp

```diff
@@ -5,122 +5,132 @@
 
 #include "Game.h"
 
-#define MUMBLE_ALLOW_DEPRECATED_LEGACY_PLUGIN_API
-#include "../mumble_legacy_plugin.h"
-
-#include "../mumble_positional_audio_utils.h"
+#include "../MumbleAPI_v_1_0_x.h"
+#include "../MumblePlugin_v_1_0_x.h"
 
+#include <cstring>
 #include <memory>
 #include <sstream>
 
-std::unique_ptr< Game > game;
+static std::unique_ptr< Game > game;
 
-static inline bool inGame(const AmongUsClient_Fields &fields) {
-	if (fields.gameMode != GameMode::FreePlay && fields.gameState != GameState::NotJoined) {
-		return true;
-	}
+mumble_error_t mumble_init(uint32_t) {
+	return MUMBLE_STATUS_OK;
+}
 
-	return false;
+void mumble_shutdown() {
 }
 
-static int fetch(float *avatarPos, float *avatarFront, float *avatarTop, float *cameraPos, float *cameraFront,
-				 float *cameraTop, std::string &context, std::wstring &identity) {
-	for (uint8_t i = 0; i < 3; ++i) {
-		avatarPos[i] = avatarFront[i] = avatarTop[i] = cameraPos[i] = cameraFront[i] = cameraTop[i] = 0.f;
-	}
+MumbleStringWrapper mumble_getName() {
+	static const char name[] = "Among Us";
 
-	const auto fields = game->clientFields();
+	MumbleStringWrapper wrapper;
+	wrapper.data           = name;
+	wrapper.size           = sizeof(name) - 1;
```

Comment:
Gotta say I consider the `strlen` approach in combination with `const char*` to be more readable in this case as at first glance the `-1` is confusing :shrug:

Reply:
I would assume it is not guaranteed but I would expect this to be trivially optimizable by any halfway decent compiler. Therefore I would assume that for practical purposes it will be performed at compile-time.
And even if not: These functions are called so infrequently that there should be no difference whatsoever :thinking: